clippy_lints/src/casts/cast_abs_to_unsigned.rs

```diff
@@ -18,25 +17,28 @@ pub(super) fn check(
     cast_to: Ty<'_>,
     msrv: Option<RustcVersion>,
 ) {
-    if_chain! {
-        if meets_msrv(msrv, msrvs::UNSIGNED_ABS);
-        if cast_from.is_integral();
-        if cast_to.is_integral();
-        if cast_from.is_signed();
-        if !cast_to.is_signed();
-        if let ExprKind::MethodCall(method_path, args, _) = cast_expr.kind;
-        if let method_name = method_path.ident.name.as_str();
-        if method_name == "abs";
-        then {
-            span_lint_and_sugg(
-                cx,
-                CAST_ABS_TO_UNSIGNED,
-                expr.span,
-                &format!("casting the result of `{}::{}()` to {}", cast_from, method_name, cast_to),
-                "replace with",
-                format!("{}.unsigned_abs()", Sugg::hir(cx, &args[0], "..")),
-                Applicability::MachineApplicable,
-            );
-        }
+    if meets_msrv(msrv, msrvs::UNSIGNED_ABS)
+        && let ty::Int(from) = cast_from.kind()
+        && let ty::Uint(to) = cast_to.kind()
+        && let ExprKind::MethodCall(method_path, args, _) = cast_expr.kind
+        && method_path.ident.name.as_str() == "abs"
```

Comment:
`sym!` would be marginally slower as it calls `Symbol::intern` which also locks + adds a hashmap lookup. I can swap it over if we prefer that style wise though

Reply:
It depends on the implementation, I can imagine that the mangling is faster due to the doc comment. It might be worth to investigate that at one point. For now, I'm okay with this, if you prefer it. :upside_down_face: